components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -884,9 +907,12 @@ void RewardsServiceImpl::OnLedgerStateLoaded(
   if (!Connected())
     return;
 
-  handler->OnLedgerStateLoaded(data.empty() ? ledger::Result::NO_LEDGER_STATE
-                                            : ledger::Result::LEDGER_OK,
-                               data);
+  ledger::Result error_code = data.empty() ? ledger::Result::NO_LEDGER_STATE
+                                           : ledger::Result::LEDGER_OK;
+  handler->OnLedgerStateLoaded(error_code, data);
+
+  TriggerOnRewardsInitialized(error_code);
```

Comment:
This doesn't look correct to me. RewardsServiceImpl should not be deciding when rewards is initialized and it shoudln't assume rewards is initialized when the ledger state is loaded. That leaks internals of the ledger library into RewardsService. This should be triggered from ledger_impl.

Reply:
I know it isn't part of this PR, but we shouldn't be calling `GetRewardsMainEnabled` below either. The LedgerClient methods in particular should not be doing anything other than handling the load/callback.

Reply:
and actually I think this is redundant and you should be using `OnWalletInitialized`

Reply:
Oh, you're right. For some reason, I was thinking that was only called when the wallet was initially created. I'll fix that, thanks for noticing!